hudi-client/src/main/java/org/apache/hudi/client/Compactor.java

```diff
@@ -45,7 +45,8 @@ public Compactor(HoodieWriteClient compactionClient) {
 
   public void compact(HoodieInstant instant) throws IOException {
     LOG.info("Compactor executing compaction " + instant);
-    JavaRDD<WriteStatus> res = compactionClient.compact(instant.getTimestamp());
+    HoodieWriteClient writeClient = compactionClient;
```

Comment:
Any specific reason for introducing a new variable here?

Reply:
This is to keep the client instance consistent between writing and committing.